### ISSUE TITLE
Support wxPython3

### ISSUE THREAD

**Issue by Cronax42** [robotframework/RIDE#1554]
It is presently impossible to use RIDE on Mac OS X. Ride requires an older version of wxPython (2.8.12.1 or thereabouts) but said version cannot be installed on Mac OS X 10.11 since the installer for said version uses some old and deprecated functionality which is no longer available.

Even manually building and installing said version of wxPython is impossible since that it contains bugs that prevent it from being installed on Mac OS X 10.11 which were only fixed for the 3.0.x versions of wxPython.

In other words, unless changes are made to RIDE to allow it to use a more current version of wxPython, it cannot be used on the latest version of Mac OS X.

**Comment by HelioGuilherme66** [comment]
You may try the pre-release package as I mentioned in PR #1537:
Hello,
We have a package for you to try, at https://github.com/HelioGuilherme66/RIDE/releases/tag/v1.6a1
In Mac OS X, there are reports of success with wxPython 2.8.12.1 but crashes with 2.9 and 3.0 versions. If you are on Windows or Linux it will be OK.
(there are some problems with GUI use, for example the Keyword help in Windows is not able to detach the window)

**Comment by Cronax42** [comment]
I am trying the version from that package but no matter what I do I can't get it to find the wxPython I installed with homebrew...As I previously mentioned, I am on OS X.

**Comment by HelioGuilherme66** [comment]
Maybe is only finding the 2.8.12.1 version (but it should use the newest).
You may confirm what version are you using with:
```
import wx
print wx.VERSION
```
You may try to use ```wx-config``` to see (or change) the version.

As a last resort, uninstall the 2.8.12.1 wxPython.
(note that I use Linux, I can't test in OSX 10.11)

**Comment by Cronax42** [comment]
I uninstalled the 2.8.12.1 version before installing the 3.0 version. wx.VERSION gives me `(3, 0, 2, 0, '')`

**Comment by HelioGuilherme66** [comment]
Please post the output or error messages from starting RIDE.
(The next thing I can only think to do is to reboot the Mac :( )

**Comment by Cronax42** [comment]
```
~/Downloads/RIDE-1.6a1/build/scripts-2.7 $ ./ride.py 
wxPython not found.
You need to install wxPython 2.8, 2.9 or 3.0 (minimum version is 2.8.12.1 with unicode support) to run RIDE.
wxPython can be downloaded from http://sourceforge.net/projects/wxpython/files/wxPython/
```

As you can see, it only tells me that it cannot find wxPython...

**Comment by HelioGuilherme66** [comment]
Another experiment is to install RIDE instead of running locally. Please do ```python setup.py install``` in ``` ~/Downloads/RIDE-1.6a1/ ```.

**Comment by Cronax42** [comment]
After installation I get the exact same error message unfortunately.

**Comment by CristhianCarmona** [comment]
I also tried to install wxPython 2.8.12.1 but it is imposssible. I got do it using this example https://www.snip2code.com/Snippet/830249/Installing-robot-framework--selenium-web/
buuuut RIDE has a lot of errors, instead of RIDE I use a plug-in for Pycharm. (My machine: macbook pro & El Capitan)

**Comment by ujan75** [comment]
I tried the provided v1.6a1 ride package with wxPython 2.8.12.1 and it didn't work for me. 

>./ride.py
wxPython not found.
You need to install wxPython 2.8, 2.9 or 3.0 (minimum version is 2.8.12.1 with unicode support) to run RIDE.
wxPython can be downloaded from http://sourceforge.net/projects/wxpython/files/wxPython/

>pip list
pip (7.1.2)
robotframework (3.0)
robotframework-ride (devel-2015-10-11)
wxPython (2.8.12.1)
wxPython-common (2.8.12.1)

**Comment by MaernFir** [comment]
exactly new ride need to be released that runs with wxpython 2.9 and 3.0 . El Capitan is not working with wxpython 2.8.12.1

**Comment by nuttasate** [comment]
I'm facing the same issue. I would like to use RIDE but it does not work on my macbook (IE Capital).

**Comment by MaernFir** [comment]
For a small period of time i used wxpython 3.0 version with ride 1.5 but when i download selenium2library files overwrited. they need to release new version of ride for el capitan

**Comment by ujan75** [comment]
I was able to use RIDE briefly on OSX Yosemite. You can try this as a work around and see if it fixes your issue.
- Installing 32 bit python
- then installing from source the wxPython 2.8.12.1 unicode
- then installing RIDE using pip

**Comment by bhajojo** [comment]
I am also using ride on mac os x.
But if at all it is not working there are other editors as well  for creating scripts

**Comment by bhajojo** [comment]
I have also worked with pycharm as a replacement for ride editor

**Comment by CristhianCarmona** [comment]
I use this plugin for textmate. It works!! 
https://bitbucket.org/jussimalinen/robot.tmbundle/wiki/Home

**Comment by Cronax42** [comment]
I am aware there are working alternatives for Ride, but that doesn't seem like a good excuse not to support Mac OS X 10.11. As mentioned, the issue is specific to El Capitan as for older version, there are workarounds possible (although they are hardly user friendly, but that is another matter...) For El Capitan however, the workaround will not work since the only versions that Ride will accept are versions that cannot be built from source on El Capitan!

**Comment by yanne** [comment]
@Cronax42: are you able to help to debug this further?

If you are, can you clone this repository and do the following

> pip install -r requirements.txt
> invoke devel

What happens?

**Comment by Cronax42** [comment]
@yanne The version in that specified branch does indeed appear to work with wxPython 3.0.2.0!

**Comment by ujan75** [comment]
@yanne This version works on Yosemite 10.10 as well with wxPython 3.0.2.0

**Comment by ujan75** [comment]
+1

**Comment by michel117** [comment]
I tested the alpha 2.01 yesterday, and was first glad to see RIDE finally works stable with wxpython 3.0.2.
Thank you Yanne and Helio !
Nevertheless, keyword completion (is this filed already? couldn't find an issue) and the not working keyboard shortcut #1441 makes it really difficult to use the Editor productively under OSX. 
And in our company Macs are predominant ;-)

**Comment by PierreLeSang** [comment]
So how to install wxpython 3.0.2. on El Capitan ?

**Comment by longmazhanfeng** [comment]
@yanne I found the alpha 2.01 works well with wxpython 3.0.2 on Mac OS X 10.11. Thank you!
Nevertheless the 1.5.2 can't work and only "easy_install" can get alpha 2.01, "pip" cannot(the latest is 1.5.2) do it. Will it be updated in Pypi later?

**Comment by neteaseknight** [comment]
@PierreLeSang, you can install wxpython in this way.
$ brew install wxpython

@yanne, @Cronax42
I install ride_v2.0.a1 + wxPython2.8.1 with "easy_install  robotframework-ride"  on OS X 10.10(Yosemite ) & 10.11(EI Capitan ), and all works.  
But the ride_v2.0.a1 have a  bug compared with ride v1.5.2.1 + wxPython2.8.1.
--------------
The **Settings** **pannels **of Direction/Suite/TestCase display inproperly in the bottom(2-3 items such as Tags/Timeout dismiss/noshow).
While it can be work-around after maxsize the ride window!
--------------

**Comment by assamuel** [comment]
Keyword completion and intellisense does not seem to work with RIDE 2.01a on El Cap. Is this an outstanding issue or has this been resolved. Without the keyword completion on OS X, it can be a tedious process to use RIDE when we have hundreds of keywords to choose from.

**Comment by ntsikelelosonjica** [comment]
Hi all. Any solution to this issue?

![ride](https://user-images.githubusercontent.com/33375312/32978086-8e03985a-cc43-11e7-9afb-59120cbbe082.png)

**Comment by HelioGuilherme66** [comment]
I am working to get RIDE on Python3, but I have not released it because there are some bugs (with workarounds). If anyone would like to help me testing (or debugging), please install from my fork, with:
```
sudo -H python3 -m pip install -U --pre https://github.com/HelioGuilherme66/RIDE/archive/python3.zip
```

But please use our Slack #ride channel, or Google Group to communicate about my fork or branch.

**Comment by HelioGuilherme66** [comment]
@Cronax42  Hey, how are you doing since 2015? ;)
This is finally fixed, so closing:
Users may want to see: [latest releases](https://github.com/robotframework/RIDE/releases)
The best installation process is:
```
pip install -r -U https://raw.githubusercontent.com/robotframework/RIDE/master/requirements.txt
pip install -U robotframework-ride
```

**Comment by teaglebuilt** [comment]
Could not find a version that satisfies the requirement Pywin32 (from robotframework-ride) (from versions: )
No matching distribution found for Pywin32 (from robotframework-ride)
